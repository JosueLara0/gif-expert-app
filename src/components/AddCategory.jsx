//* libraries
import { useState } from "react";
import PropTypes from "prop-types";

//------------------------------------------------------------------------------
export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  //* First way
  //   const onInputChange = (event) => setInputValue(event.target.value);
  //* Second way
  const onInputChange = ({ target }) => setInputValue(target.value);

  const onSubmit = (event) => {
    event.preventDefault();
    const inputValueTrim = inputValue.trim();
    // If the length of the element is '<= 1' then is not added

    console.log(inputValueTrim);
    if (inputValueTrim.length <= 1) return;

    //* First way
    onNewCategory(inputValueTrim);
    //* Second way
    // setCategories((categories) => [inputValueTrim, ...categories]);
    setInputValue("");
  };

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Search GIFs"
        value={inputValue}
        //* Firs way
        //   onChange={(event) => onInputChange(event)}
        //* Second way
        onChange={onInputChange}
      />
    </form>
  );
};

//? Props validations
AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
