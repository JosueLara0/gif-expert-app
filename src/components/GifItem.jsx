//* libraries
import PropTypes from "prop-types";

//------------------------------------------------------------------------------
export const GifItem = ({ title, url }) => {
  return (
    <div className="card">
      <img src={url} alt={title} loading="lazy" />
      <p>{title}</p>
    </div>
  );
};

//? Props validations
GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
