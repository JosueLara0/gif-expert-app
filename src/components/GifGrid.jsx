//* libraries
import PropTypes from "prop-types";
//* components
import { GifItem } from "./GifItem";
//* hooks
import { useFetchGifs } from "../hooks/useFetchGifs";

//------------------------------------------------------------------------------
export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>

      {isLoading && <h2>Cargando...</h2>}

      <div className="card-grid">
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};

//? Props validations
GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
