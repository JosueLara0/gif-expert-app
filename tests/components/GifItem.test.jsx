//* libraries
import { render, screen } from "@testing-library/react";
//* components
import { GifItem } from "../../src/components/GifItem";

//? Tests
describe("Tests in <GifItem/>", () => {
  const title = "One Punch";
  const url =
    "https://media2.giphy.com/media/oxbNORcXx76F2/giphy.gif?cid=aa030f39gigshg8jcumtz662xsloyek0d7jblzpwqvqnxuhh&rid=giphy.gif&ct=g";

  test("should make match with the snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("should show img with the right url and alt", () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug();
    // expect(screen.getByRole("img").src).toBe(url);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });

  test("should show title in the component", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
