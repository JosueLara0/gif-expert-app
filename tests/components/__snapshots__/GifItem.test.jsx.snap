// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifItem/> should make match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch"
      loading="lazy"
      src="https://media2.giphy.com/media/oxbNORcXx76F2/giphy.gif?cid=aa030f39gigshg8jcumtz662xsloyek0d7jblzpwqvqnxuhh&rid=giphy.gif&ct=g"
    />
    <p>
      One Punch
    </p>
  </div>
</div>
`;

exports[`Tests in <GifItem/> should make match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch"
      loading="lazy"
      src="https://media2.giphy.com/media/oxbNORcXx76F2/giphy.gif?cid=aa030f39gigshg8jcumtz662xsloyek0d7jblzpwqvqnxuhh&rid=giphy.gif&ct=g"
    />
    <p>
      One Punch
    </p>
  </div>
</div>
`;
